apache/couchdb-fauxton
# Mango query UX improvements

## Overview

A few adjustments that should improve UX when coding Mango queries in Fauxton:

- Set code editor max height so buttons and stats are not hidden when the code spans several lines.

Before:
<img width="503" alt="image" src="https://github.com/apache/couchdb-fauxton/assets/30349380/36353eaa-2943-4cac-bf06-bf4560492fbd">

After:
<img width="499" alt="image" src="https://github.com/apache/couchdb-fauxton/assets/30349380/ad8394c2-96fd-4ad4-b855-1a247768169c">

- Add cheatsheet modal that shows query operators.

<img width="1162" alt="image" src="https://github.com/apache/couchdb-fauxton/assets/30349380/4f7da3e4-3399-4cd6-9785-08cb40bdeb0c">

- Display exec stats and warning directly below the editor instead of in a popover to give easier access to that information, particularly when trying to optimize queries.

Before:
<img width="434" alt="image" src="https://github.com/apache/couchdb-fauxton/assets/30349380/34e93ace-0137-47cf-bc12-f47bfa693c38">


After:
<img width="438" alt="image" src="https://github.com/apache/couchdb-fauxton/assets/30349380/147e4356-d60c-42bb-bf81-889a422127b4">


## Testing recommendations

In the 'Run a Query with Mango' page:
- Run queries and verify exec stats and warning are displayed
- Verify cheatsheet modal opens when clicking on the new icon added

## GitHub issue number

n/a

## Related Pull Requests

n/a

## Checklist

- [ ] Update [rebar.config.script](https://github.com/apache/couchdb/blob/main/rebar.config.script) with the correct tag once a new Fauxton release is made
